### PR TITLE
Check if kernel image is readable before mounting vm

### PR DESCRIPTION
**Pull request by magalilemes**
If the kernel image isn't readable, guestmount throws a few errors, and the virtual machine cannot be mounted. Here we check if the kernel image is readable, and if it isn't we ask the user to give it read permission.

### CODE REVIEW

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

File: src/vm.sh [hunk 1]
```diff
@@ -18,6 +18,14 @@ function vm_mount()
   qemu_img_path="${qemu_img_path:-${configurations[qemu_path_image]}}"
   mount_point_path="${mount_point_path:-${configurations[mount_point]}}"
 
+  if [[ ! -r "/boot/vmlinuz-$(uname -r)" ]]; then
+    say 'To mount the VM, the kernel image needs to be readable'
+    if [[ $(ask_yN "Run 'sudo chmod +r /boot/vmlinuz-$(uname -r)'?") =~ 0 ]]; then
+      return 125 # ECANCELED
+    fi
+    cmd_manager "$flag" "sudo chmod +r /boot/vmlinuz-$(uname -r)"
```
[the remark this answers is not in the document]
Comment (magalilemes) [reply src/vm.sh:26]: > ```
> curl $(python3 -c "import apt, os; print(apt.cache.Cache()['linux-image-' + os.uname().release].versions[0].uri)") | dpkg-deb -x - .
> ```
@rodrigosiqueira I think this one command is not meant for making the kernel image readable. Can you enlighten me on what it does? It gets a kernel image and then extracts it with dpkg-deb?


> Also, it looks like that the correct command should be something like this:
> 
> ```
> sudo dpkg-statoverride --update --add root root 0644 /boot/vmlinux-$(uname -r)
> ```
This one works too!!

**Comment by magalilemes** [comment]
I couldn't really find an alternative that didn't require `sudo`. Should  `sudo dpkg-statoverride --update --add root root 0644 /boot/vmlinux-$(uname -r)` be used or the command from the current code is fine?
I'll add tests as soon as this is solved.

**Comment by rodrigosiqueira** [comment]
> I couldn't really find an alternative that didn't require `sudo`. Should `sudo dpkg-statoverride --update --add root root 0644 /boot/vmlinux-$(uname -r)` be used or the command from the current code is fine?
> I'll add tests as soon as this is solved.

Ok, if there is no other way, do you know the advantage of using the above command compared with chown?

File: src/vm.sh [hunk 2]
```diff
@@ -18,6 +20,14 @@ function vm_mount()
   qemu_img_path="${qemu_img_path:-${configurations[qemu_path_image]}}"
   mount_point_path="${mount_point_path:-${configurations[mount_point]}}"
 
+  if [[ ! -r "${prefix}boot/vmlinuz-$(uname -r)" ]]; then
+    say 'To mount the VM, the kernel image needs to be readable'
+    if [[ $(ask_yN 'Do you want to make your host kernel image readable?') =~ 0 ]]; then
+      return 125 # ECANCELED
+    fi
+    cmd_manager "$flag" "sudo dpkg-statoverride --update --add root root 0644 ${prefix}boot/vmlinuz-$(uname -r)"
```
Comment (rodrigosiqueira) [inline src/vm.sh:28]: This command will only work on Debian-based systems; check if the host system is Debian-based before you use this command. If it is not a Debian system, use the standard chmod.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
I just have one final observation. Could you address that? Do not forget to update the test with my latest suggestion.

**Comment by rodrigosiqueira** [comment]
Hi @magalilemes ,
Thanks a lot for fixing this issue. Merged to the unstable branch.

**Pull request closed without merging**